Libraries/Contents/ContentExtensions.cs

```diff
@@ -86,5 +89,29 @@ public static PublicationStatus GetPublicationStatus(this IContent content)
             if (content.ContentItem.Published) return PublicationStatus.Published;
             return content.ContentItem.Latest ? PublicationStatus.Draft : PublicationStatus.Deleted;
         }
+
+        /// <summary>
+        /// Prevents multiple "latest" versions in case somehow two threads edited the same <see cref="ContentItem"/> at
+        /// the same time. For example this is possible if the update was done through XHR.
+        /// </summary>
+        /// <param name="content">The desired latest version of the content.</param>
+        public static async Task SanitizeSurveyVersionsAsync(this IContent content, ISession session)
+        {
+            var contentItemId = content.ContentItem.ContentItemId;
+            var contentItemVersionId = content.ContentItem.ContentItemVersionId;
+            var stuckOtherDocuments = (await session
+                    .Query<ContentItem, ContentItemIndex>(index =>
+                        (index.Latest || index.Published) &&

```

Comment:
Hmm, szerintem ezeket AND-elni kellene. Ugyanis olyan lehet, hogy egy content itemnek van másik Latest vagy Published verziója. Ugye pl. ha draftot hozol létre egy published itemhez, akkor az lesz a Latest de nem Published, a korábbi verzió marad a Published, de már nem lesz Latest.

Továbbá, ehhez kapcsolódóan raknék ide valami ellenőrzést a bemeneti itemre. Ha mondjuk az biztos, hogy Latest AND Published, akkor ez oké.

Reply:
Ha draftot hozol létre akkor azzal unpublish az előző verziót, nem? Mert akkor azt kéne csinálni, hogy csak akkor fut le, ha `content.Latest == true` és akkor a queryben is csak `index.Latest` az `index.Latest || index.Published` helyett.

Reply:
Nem, draft lehet úgy is, hogy a korábbi marad published (ugye egy publishelt itemet így tudsz draftként szerkeszteni anélkül, hogy publicban bármi látszana).

Igen, úgy jó a logika.